src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -334,6 +400,31 @@ def execute_link(
             else:
                 cropped_path = nso_manipulator.add_height(cropped_path, add_height_band)
 
+        if fill_with_nearest_date:
+            print(
+                "-----Filling satellite image with data from the nearest other satellite----"
+            )
+            nearest_georegion = nso_georegion(
+                coordinates=fill_coordinates,
+                previous_link=link,
+                output_folder=self.output_folder,
+                username=self.username,
+                password=self.password,
+            )
+
+            nearest_link = nearest_georegion.retrieve_download_links(
+                find_nearest_to_previous_link=True
+            )
+
+            cropped_path_fill = nearest_georegion.execute_link(nearest_link[0])
```

Comment:
I think we should be careful with simply downloading more and more data and be a bit smarter with this. (as these files tend to be dozens of Gigabytes large and have quite some download time).

Reply:
How would you solve this without downloading? execute_link  already searches for .tif files and .zip if they have been downloaded before so to avoid the downloading otherwise

Reply:
True, I forgot it already checks for existing files!